src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -720,6 +720,14 @@ public function getSession()
         $session = $this->session;
         if (!$session instanceof SessionInterface && null !== $session) {
             $this->setSession($session = $session());
+            /*
+             * For supporting sessions in php runtime with runners like roadrunner or swoole the session
+             * cookie need read from the cookie bag and set on the session storage.
+             */
+            if (!$session->isStarted()) {
```

Comment:
Any particular reason to call this code only when the injected session is a factory? This sounds "fragile" to me.

Reply:
Wanted to avoid that it is called every time `getSession()` is triggered. Maybe I should just check if already a session id was set else read it from the cookie, what do you think?